# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../shadow/service-domain-shadow.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Well - we want all the shadow stuff groupped into the "service / security
domain" shadow; as it was up to 4.0.3 (or maybe even slightly higher versions).

Mostly:

sed -i 's/pam_start *("[^l"]*"/pam_start ("shadow"/g' *.c

  - Rene Rebe <rene@exactcode.de>

diff -ur shadow-4.0.7/src/chpasswd.c shadow-4.0.7-fixed/src/chpasswd.c
--- shadow-4.0.7/src/chpasswd.c	2005-01-18 00:02:41.000000000 +0100
+++ shadow-4.0.7-fixed/src/chpasswd.c	2005-02-14 23:16:02.391210464 +0100
@@ -156,7 +156,7 @@
 
 	if (retval == PAM_SUCCESS) {
 		retval =
-		    pam_start ("chpasswd", pampw->pw_name, &conv, &pamh);
+		    pam_start ("shadow", pampw->pw_name, &conv, &pamh);
 	}
 
 	if (retval == PAM_SUCCESS) {
diff -ur shadow-4.0.7/src/groupadd.c shadow-4.0.7-fixed/src/groupadd.c
--- shadow-4.0.7/src/groupadd.c	2005-01-18 00:02:41.000000000 +0100
+++ shadow-4.0.7-fixed/src/groupadd.c	2005-02-14 23:16:02.424205448 +0100
@@ -513,7 +513,7 @@
 
 	if (retval == PAM_SUCCESS) {
 		retval =
-		    pam_start ("groupadd", pampw->pw_name, &conv, &pamh);
+		    pam_start ("shadow", pampw->pw_name, &conv, &pamh);
 	}
 
 	if (retval == PAM_SUCCESS) {
diff -ur shadow-4.0.7/src/groupmod.c shadow-4.0.7-fixed/src/groupmod.c
--- shadow-4.0.7/src/groupmod.c	2005-01-18 00:02:41.000000000 +0100
+++ shadow-4.0.7-fixed/src/groupmod.c	2005-02-14 23:16:02.439203168 +0100
@@ -502,7 +502,7 @@
 
 	if (retval == PAM_SUCCESS) {
 		retval =
-		    pam_start ("groupmod", pampw->pw_name, &conv, &pamh);
+		    pam_start ("shadow", pampw->pw_name, &conv, &pamh);
 	}
 
 	if (retval == PAM_SUCCESS) {
diff -ur shadow-4.0.7/src/newusers.c shadow-4.0.7-fixed/src/newusers.c
--- shadow-4.0.7/src/newusers.c	2005-01-18 00:02:41.000000000 +0100
+++ shadow-4.0.7-fixed/src/newusers.c	2005-02-14 23:16:02.511192224 +0100
@@ -344,7 +344,7 @@
 
 	if (retval == PAM_SUCCESS) {
 		retval =
-		    pam_start ("newusers", pampw->pw_name, &conv, &pamh);
+		    pam_start ("shadow", pampw->pw_name, &conv, &pamh);
 	}
 
 	if (retval == PAM_SUCCESS) {
diff -ur shadow-4.0.7/src/useradd.c shadow-4.0.7-fixed/src/useradd.c
--- shadow-4.0.7/src/useradd.c	2005-01-18 00:02:41.000000000 +0100
+++ shadow-4.0.7-fixed/src/useradd.c	2005-02-14 23:16:02.582181432 +0100
@@ -1651,7 +1651,7 @@
 
 	if (retval == PAM_SUCCESS) {
 		retval =
-		    pam_start ("useradd", pampw->pw_name, &conv, &pamh);
+		    pam_start ("shadow", pampw->pw_name, &conv, &pamh);
 	}
 
 	if (retval == PAM_SUCCESS) {
diff -ur shadow-4.0.7/src/usermod.c shadow-4.0.7-fixed/src/usermod.c
--- shadow-4.0.7/src/usermod.c	2005-01-18 00:02:41.000000000 +0100
+++ shadow-4.0.7-fixed/src/usermod.c	2005-02-14 23:16:02.613176720 +0100
@@ -1404,7 +1404,7 @@
 
 	if (retval == PAM_SUCCESS) {
 		retval =
-		    pam_start ("usermod", pampw->pw_name, &conv, &pamh);
+		    pam_start ("shadow", pampw->pw_name, &conv, &pamh);
 	}
 
 	if (retval == PAM_SUCCESS) {
--- shadow-4.0.7/src/groupdel.c	2005-01-18 00:02:41.000000000 +0100
+++ shadow-4.0.7-fixed/src/groupdel.c	2005-02-14 23:17:50.432785648 +0100
@@ -309,7 +309,7 @@
 
 	if (retval == PAM_SUCCESS) {
 		retval =
-		    pam_start ("groupdel", pampw->pw_name, &conv, &pamh);
+		    pam_start ("shadow", pampw->pw_name, &conv, &pamh);
 	}
 
 	if (retval == PAM_SUCCESS) {
